TypewiseAlert.Test/TypewiseAlertTest.cs

```diff
@@ -1,15 +1,143 @@
 using System;
 using Xunit;
+using TypewiseAlert.Notifiers;
+using static TypewiseAlert.TypewiseAlert;
 
 namespace TypewiseAlert.Test
 {
   public class TypewiseAlertTest
   {
+
+    [Fact]
+    public void InfersBreachForLowLimit()
+    {
+        Assert.True(InferBreach(12, 20, 30) == BreachType.TOO_LOW);
+    }
+
+    [Fact]
+    public void InfersBreachForHighLimit()
+    {
+        Assert.True(InferBreach(35, 20, 30) == BreachType.TOO_HIGH);
+    }
+
+    [Fact]
+    public void InfersBreachForNormal()
+    {
+        Assert.True(InferBreach(22, 20, 30) == BreachType.NORMAL);
+    }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForPassiveCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -5) == BreachType.TOO_LOW);
+    }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForMediumCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 45) == BreachType.TOO_HIGH);
+    }
+
+    [Fact]
+    public void ClassifyTemperatureBreachForHighCooling()
+    {
+        Assert.True(ClassifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 42) == BreachType.NORMAL);
+    }
+    
+    [Fact]
+    public void ConsoleCheckAndAlertTest()
+    {
+        var _notifierType = new FakeConsoleNotifier();
+        CheckAndAlert(_notifierType, new BatteryCharacter { brand = "XYZ", coolingType = CoolingType.MED_ACTIVE_COOLING }, -5);
+        Assert.True(_notifierType.IsConsoleTriggerNotificationCalled);
+    }
+
+    [Fact]
+    public void EmailCheckAndAlertTest()
+    {
+        var _notifierType = new FakeEmailNotifier();
+        CheckAndAlert(_notifierType, new BatteryCharacter { brand = "ABC", coolingType = CoolingType.PASSIVE_COOLING }, 50);
+        Assert.True(_notifierType.IsEmailTriggerNotificationCalled);
+    }
+
+    [Fact]
+    public void ControllerCheckAndAlertTest()
+    {
+        var _notifierType = new FakeControllerNotifier();
+        CheckAndAlert(_notifierType, new BatteryCharacter { brand = "PQR", coolingType = CoolingType.HI_ACTIVE_COOLING }, -5);
+        Assert.True(_notifierType.IsControllerTriggerNotificationCalled);
+    }
+
+    [Fact]
+    public void CompositeCheckAndAlertTest()
```

Comment:
Hi split the compositeCheckAndAlertTest() to multiple independent test functions ...remember when to use multiple assertions

Reply:
Created multiple independent test functions for the CheckAndAlert of Composite alert. CompostiteCheckAndAlertTest() will test on the communication between CheckAndAlert() and Composite TriggerNotification(). CompositeNotificationTest() will test on the multiple alerts i.e., console and email TriggerNotfication() calls.